jetty-ee10/jetty-ee10-servlet/src/main/java/org/eclipse/jetty/ee10/servlet/ServletContextHandler.java

```diff
@@ -830,22 +830,19 @@ public Set<String> getResourcePaths(String path)
         {
             Resource resource = getResource(path);
 
-            if (resource != null && resource.exists())
-            {
-                if (!path.endsWith(URIUtil.SLASH))
-                    path = path + URIUtil.SLASH;
+            if (!path.endsWith(URIUtil.SLASH))
+                path = path + URIUtil.SLASH;
+
+            HashSet<String> set = new HashSet<>();
 
-                List<String> l = resource.list();
-                if (l != null)
+            for (Resource r: resource)
```

Comment:
No need for this loop, as `list()` takes care of that if the resource is a composite.

Reply:
Done